### PR TITLE
Update test runner

### PR DESCRIPTION
**Pull request by mymedia2**

### CODE REVIEW

**Review by SirVer — COMMENTED** [review]
Thanks for your contribution! A few comments from my site to consider.

File: test/vim_interface.py [hunk 1]
```diff
@@ -174,9 +169,9 @@ def _send(self, s):
         s = s.replace(";", r"\;")
 
         if len(s) == 1:
-            silent_call(["tmux", "send-keys", "-t", self.session, hex(ord(s))])
+            subprocess.check_call(["tmux", "send-keys", "-t", self.session, hex(ord(s))])
```
[the remark this answers is not in the document]
Comment (mymedia2) [reply test/vim_interface.py:172]: Yes, "tmux send-keys" may print messages to stderr. For example, if a tmux session was not started or if the TERM environment variable has unusual value. In such a case, one who runs the tests, will see the error immediately and then a stack trace from `subprocess.check_call`. Before my change the tests hang here.

On success, "tmux send-keys" prints nothing.

File: test_all.py [hunk 2]
```diff
@@ -200,14 +200,21 @@ def main():
...
 
-        vim = None
-        vim_flavor = "vim"
         if options.interface == "tmux":
             vim = VimInterfaceTmux(options.vim, options.session)
-            vim_flavor = "vim"
         else:
             vim = VimInterfaceTmuxNeovim(options.vim, options.session)
+
+        has_nvim = subprocess.check_output(
```
[the remark this answers is not in the document]
Comment (mymedia2) [reply test_all.py:208]: On second thoughts, it looks like part of version number which is autodetected. Perhaps, it is worth to combine the checks. And print a warning if unsupported version is found.

File: test/vim_interface.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SirVer) [reply test/vim_interface.py:172]: Ah, great! So this is an improvement. Can you add this information as a comment into the code?

File: test_all.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SirVer) [reply test_all.py:208]: Sorry, I do not follow your comment. Can you rephrase?

**Comment by SirVer** [comment]
I wanted to get this over the finish line today and realized I cannot edit the code in this PR. I took the liberty of forking this into #1491, modify it slightly and merge it there. Thanks for your contributions!
